src/connection/query.ts

```diff
@@ -12,28 +12,57 @@ export class Query<T> {
   private correlationId: string;
   private sql: string;
   private isPrepared: boolean = false;
-  private parameters: any[];
+  private parameters: any[]|undefined;
   private rowsToFetch: number = 100;
   private isCLCommand: boolean;
   private state: QueryState = QueryState.NOT_YET_RUN;
 
-  constructor(private job: SQLJob, query: string, opts: QueryOptions = {isClCommand: false, parameters: undefined}) {
+  public autoClose: boolean | undefined;

```

Comment:
An undefined boolean seems unconventional, and you always define it in the constructor, so I think the `| undefined` here can go away (?)

Reply:
Undefined boolean absolutely seems odd. But since `QueryOptions.autoClose` is `boolean?` (the same as `boolean|undefined`), I did it here. But you are right, both `false` and `undefined` are falsey values.

Reply:
resolved at https://github.com/halcyon-tech/vscode-db2i/pull/66/commits/96d3a4b5e79ac1c0c3bd5d082b2bc52406496a96